### ISSUE TITLE
FR: a helper function to update a fork

### ISSUE THREAD

**Issue by cderv** [r-lib/usethis#575]
Starting to contribute on tidyverse dev day, I find myself in the situation of updating my existing fork with last changes.  

I don't know what is your worflow but mine is opening the terminal and 
```sh
git checkout master
git pull upstream master
git push origin master
```

and now I am ready to go (using the `pr_*` functions I guess)

Would you find it helpful to have a function in usethis for updating fork like this ? 
Seems like a helper for PR contribution

**Comment by batpigandme** [comment]
@jimhester?

**Comment by jimhester** [comment]
Yeah I think this makes sense to have, not sure what it should be called, maybe `git_update`?

**Comment by jimhester** [comment]
Also I am not sure it should push the changes back to GH by default...

**Comment by cderv** [comment]
it could be an argument defaulting to FALSE or maybe a usethis-like question asking if we want to update ? 

If it is seen as a helper fir contribution it could be `fork_update` , with a default use of upstream remote ?

**Comment by hadley** [comment]
This is somehow tied to PRs, so I'd prefer it to be `pr_something()`. Or maybe `pr_init()` should call it automatically? And `check_branch_current()` should suggest it?

Related to #590

**Comment by hadley** [comment]
After #607, I think `check_branch_current()` should work correctly with forked master, and it just remains to write a function to actually update the branch.

**Comment by hadley** [comment]
@jennybc I don't think we need `pr_sitrep()` to check if you're in a good position to initiate a PR, because `pr_init()` checks everything. But you might need it during the PR process — it could tell you if you have commits locally that aren't in the PR (and vice versa), and if there's code in upstream/master that you don't have locally.  I'll move that into a separate issue.

**Comment by hadley** [comment]
We ended up calling it `pr_pull_source()`
